js/components/graph/Sidebar.js

```diff
@@ -5,14 +5,25 @@ export default class Sidebar extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      collapsed: true
+      collapsed: true,
+      results: []
     };
   }
 
   toggleSidebar = () => {
       this.setState({ collapsed: !this.state.collapsed });
   }
 
+  filteredSearch = (posts, query) => {
+    if (!query || !posts) {
+      return;
+    }
+
+    return posts.filter((post) => {
+      return post.includes(query);
```

Comment:
Convert both post and query to upper case. Also, post is not a good name for this argument. Something like `course` would be a lot better.